src/main/resources/assets/computercraft/lua/rom/apis/parallel.lua

```diff
@@ -14,7 +13,40 @@ local function create( ... )
     return tCos
 end
 
-local function runUntilLimit( _routines, _limit )
+local external = {
+    __index = {
+        add = function(bundle, ...)
+            local set = create(1, ...)
+            local rfuncs = {}
+            for i = 1, #set do
+                bundle[#bundle+1] = set[i]
+                rfuncs[#rfuncs+1] = function()
+                    local del = #bundle
+                    return table.remove(bundle, del) and true
+                end
+            end
+            return table.unpack(rfuncs)
+        end,
+        stop = function(bundle)
+            bundle.data.run = false
+        end
+    }
+}
+
+
+local function setup( bundle, ... )
```

Comment:
It feels a little odd to have `setup` either take a bundle or list of functions. I'd be inclined to have the `wait*` functions as methods on the bundle (instead of doing `parallel.waitForAny(bundle)`, have `bundle:waitForAny()`), and then just keep the original `waitFor*` methods as-is.

Reply:
ooh good idea, that'll be the next step I take.